src/main/java/de/tum/in/www1/artemis/service/plagiarism/PlagiarismService.java

```diff
@@ -25,50 +15,17 @@
 
     private static final String OTHER_SUBMISSION = "Other submission";
 
-    private final ExerciseRepository exerciseRepository;
-
-    private final PlagiarismResultRepository plagiarismResultRepository;
-
     private final PlagiarismComparisonRepository plagiarismComparisonRepository;
 
-    public PlagiarismService(ExerciseRepository exerciseRepository, PlagiarismResultRepository plagiarismResultRepository,
-            PlagiarismComparisonRepository plagiarismComparisonRepository) {
-        this.exerciseRepository = exerciseRepository;
-        this.plagiarismResultRepository = plagiarismResultRepository;
+    public PlagiarismService(PlagiarismComparisonRepository plagiarismComparisonRepository) {
         this.plagiarismComparisonRepository = plagiarismComparisonRepository;
     }
 
     /**
-     * Collects all plagiarism cases for a given course
-     *
-     * @param courseId of the course
-     * @return the collected plagiarism cases
-     */
-    public List<PlagiarismCaseDTO> collectAllPlagiarismCasesForCourse(Long courseId) {
-        // TODO why do we do this so strangely (this is working legacy code)? Refactor in a follow up
-        var collectedPlagiarismCases = new ArrayList<PlagiarismCaseDTO>();
-        var exerciseIDs = exerciseRepository.findAllIdsByCourseId(courseId);
-        exerciseIDs.forEach(id -> {
-            var exerciseOptional = exerciseRepository.findById(id);
-            if (exerciseOptional.isPresent()) {
-                PlagiarismResult<?> result = plagiarismResultRepository.findFirstByExerciseIdOrderByLastModifiedDateDescOrNull(exerciseOptional.get().getId());
-                if (result != null) {
-                    Set<PlagiarismComparison<?>> filteredComparisons = result.getComparisons().stream().filter(c -> c.getStatus() == PlagiarismStatus.CONFIRMED)
-                            .collect(Collectors.toSet());
-                    if (filteredComparisons.size() > 0) {
-                        collectedPlagiarismCases.add(new PlagiarismCaseDTO(exerciseOptional.get(), filteredComparisons));
-                    }
-                }
-            }
-        });
-        return collectedPlagiarismCases;
-    }
-
-    /**
-     * Anonymizes the comparison for the student view.
+     * Anonymize the comparison for the student view.
      * A student should not have sensitive information (e.g. the userLogin of the other student)
```

Comment:
There seems to be a problem with anonymization. During my test, the comparison view was not anonymized, and if I have a look at the screenshots of @canberkanar, this did not seem to be the case for him either. 
![image](https://user-images.githubusercontent.com/80622272/156351219-10e41bf2-8a45-4396-9239-0329b01b10a2.png)

Reply:
@willergeorg The comparisons should be anonymized again :)